Assets/Mirror/Examples/Chat/Scripts/Player.cs

```diff
@@ -8,8 +8,8 @@ public class Player : NetworkBehaviour
     {
         [SyncVar]
         public string playerName;
-
-        public ChatWindow chatWindow => ((ChatNetworkManager)NetworkManager.singleton).chatWindow;
+        
+        public ChatWindow chatWindow => ((ChatNetworkManager)FindObjectOfType<NetworkManager>()).chatWindow;
```

Comment:
FindObjectOfType is arguably worse than the previous solution.
of course, this is just in one of our examples.
but it raises the question, how would existing projects find the networkmanager in similar situations? findobjectoftype is not an option in bigger projects.

Reply:
I agree with vis,  FindObjectOfType is just as bad as singletons

The Player object has a reference to the networkClient,  and we probably need a reference to the network manager from the networkclient

Reply:
You're right, I need to find a solution about it.
@paulpach But wouldn't it increase circular dependency?

Reply:
@Lymdun yes,  you are right,  it woudl be best if we can avoid that circular dependency.  

Now,  in this particular case,  we don't need the networkmanager at all,  all we need is the chatwindow.

Reply:
What about keeping a rolling cache of the chat text in Player.cs. The UI needs to know about the script. The script doesn't need to know about the UI. Then it would also function without UI for possible testing?

Reply:
yeah,  that might be better @uweenukr

Reply:
Fixed